src/engine/BMDie.php

```diff
@@ -290,7 +290,13 @@ public function roll($isTriggeredByAttack = FALSE, $isSubdie = FALSE) {
 
         if (!isset($this->value) ||
             ($this->doesReroll && !$this->has_flag('JustPerformedTripAttack'))) {
-            $this->set__value(bm_rand($this->min, $this->max));
+            $hookResultArray = $this->run_hooks(__FUNCTION__, array('die' => $this));
+
+            // if all hook results are FALSE, then roll the die
+            if (empty($hookResultArray) ||
+                (0 == count(array_filter($hookResultArray, function ($value) { return $value !== FALSE; })))) {
+                $this->set__value(bm_rand($this->min, $this->max));
+            }
```

Comment:
Ah sorry, got the response around the wrong way.

A TRUE response from the die hook says that the die already has a die value set (which happens to be the maximum value), and thus the following bit within the if statement doesn't run.